microcosm_flask/tests/test_paging.py

```diff
@@ -42,12 +44,19 @@ def test_offset_limit_page_to_from_dict():
 
 def test_offset_limit_page_from_query_string():
     graph = create_object_graph(name="example", testing=True)
-    with graph.flask.test_request_context(query_string="offset=1&foo=bar"):
+    with graph.flask.test_request_context(query_string="offset=1"):
         page = OffsetLimitPage.from_query_string(OffsetLimitPageSchema())
         assert_that(page.offset, is_(equal_to(1)))
         assert_that(page.limit, is_(equal_to(20)))
-        # schema filters out extra arguments
-        assert_that(page.to_dict(), is_not(has_entry("foo", "bar")))
+
+
+def test_offset_limit_page_rejects_unknown_queries():
+    graph = create_object_graph(name="example", testing=True)
+    with graph.flask.test_request_context(query_string="offset=1&foo=bar"):
+        assert_that(
+            calling(OffsetLimitPage.from_query_string).with_args(OffsetLimitPageSchema()),
+            raises(ValidationError),
```

Comment:
Note: This is a *breaking* change; we now validate all incoming query arguments explicitly.

Reply:
oh..... Interesting. I feel like this might break things in places. Does that mean this will start to fail in unit tests too? Do we need to bump marshmallow in `marquez` or something?

Reply:
That's a good question. The thing is that I think we actually do want errors on invalid parameters. Think of how many bugs there are when you pass in a parameter you think should exist, but actually doesn't.

Reply:
I agree that this is a good thing.